index.d.ts

```diff
@@ -1,6 +1,16 @@
-export default function CompassDigitalId(
-  data: string,
-  provider: string,
-  type: string,
-  id: string
-): string;
+declare module '@compassdigital/id' {
+	export type CdlEncodedID = string;
+
+	export interface CdlDecodedID {
+		service: string;
+		provider: string;
+		type: string;
+		id: string;
+	}
+
+	function ID(service: string, provider: string, type: string, id: string): CdlEncodedID;
+	function ID(decoded: CdlDecodedID): CdlEncodedID;
+	function ID(encoded: CdlEncodedID): CdlDecodedID;
```

Comment:
this could also return undefined if CdlEncondedID is not a valid id

Reply:
Good catch!